Scripts/RemediationScripts/Remediate-Azure_AppService_DP_Use_Secure_FTP_Deployment.ps1

```diff
@@ -0,0 +1,1132 @@
+        App Services should use secure FTP deployments.
+        
+        .PARAMETER SubscriptionId
+        Specifies the ID of the Subscription to be remediated.
+        
+        .PARAMETER Force
+        Specifies a forceful remediation without any prompts.
+        
+        .Parameter PerformPreReqCheck
+        Specifies validation of prerequisites for the command.
+        
+        .PARAMETER DryRun
+        Specifies a dry run of the actual remediation.
+        
+        .PARAMETER FilePath
+        Specifies the path to the file to be used as input for the remediation.
+
+        .PARAMETER FTPSOnly
+        Specifies the FTP State to be used as input for the remediation.
+
+        .PARAMETER DisableFTP
+        Specifies the FTP State to be used as input for the remediation.
+
+        .INPUTS
+        None. You cannot pipe objects to Enable-SecureFTPDeploymentForAppServices.
+
+        .OUTPUTS
+        None. Enable-SecureFTPDeploymentForAppServices does not return anything that can be piped and used as an input to another command.
+
+        .EXAMPLE
+        PS> Enable-SecureFTPDeploymentForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun 
+
+        .EXAMPLE
+        PS> Enable-SecureFTPDeploymentForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FTPSOnly
+
+        .EXAMPLE
+        PS> Enable-SecureFTPDeploymentForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202201011212\EnableFTPSOnlyForAppServices\AppServicesWithFTPSOnlyEnabled.csv
+
+        .LINK
+        None
+    #>
+
+    param (
+        [String]
+        [Parameter(ParameterSetName = "DryRun", Mandatory = $true, HelpMessage="Specifies the ID of the Subscription to be remediated")]
+        [Parameter(ParameterSetName = "WetRun", Mandatory = $true, HelpMessage="Specifies the ID of the Subscription to be remediated")]
+        $SubscriptionId,
+
+        [Switch]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies a forceful remediation without any prompts")]
+        $Force,
+
+        [Switch]
+        [Parameter(ParameterSetName = "DryRun", HelpMessage="Specifies validation of prerequisites for the command")]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies validation of prerequisites for the command")]
+        $PerformPreReqCheck,
+
+        [Switch]
+        [Parameter(ParameterSetName = "DryRun", Mandatory = $true, HelpMessage="Specifies a dry run of the actual remediation")]
+        $DryRun,
+
+        [String]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies the path to the file to be used as input for the remediation")]
+        $FilePath,
+
+        [Switch]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies the FTP State to be configured")]
```

Comment:
Two separate params for same settings would create confusion, let's have one param with two allowed values 'FTPS' or 'Disabled'

Reply:
Used one parameter FTPState for both as mentioned above.